### PR TITLE
Add name with change info to segment changes (1-1230-update-name-change)

### PR DESCRIPTION
**Pull request by thomasheartman**
Extract and reuse the component that we use for strategy name changes.

## Discussion points:

This impl only shows the new name in the second field (editing segment). Should we show it in both? Why do we have both? Do we have UI designs for this?

image

### CODE REVIEW

File: frontend/src/component/changeRequest/ChangeRequest/NameWithChangeInfo/NameWithChangeInfo.test.tsx
Comment (thomasheartman) [inline frontend/src/component/changeRequest/ChangeRequest/NameWithChangeInfo/NameWithChangeInfo.test.tsx:1]: This is just the same test file moved and updated with new arg names.

File: frontend/src/component/changeRequest/ChangeRequest/NameWithChangeInfo/NameWithChangeInfo.tsx
Comment (thomasheartman) [inline frontend/src/component/changeRequest/ChangeRequest/NameWithChangeInfo/NameWithChangeInfo.tsx:1]: The extracted component. I see we use `Truncated` a number of places around change requests. Maybe it's worth defining once and used as a dependency?

File: frontend/src/component/changeRequest/ChangeRequest/SegmentTooltipLink.tsx [hunk 1]
```diff
@@ -70,9 +71,10 @@ export const SegmentTooltipLink: FC<IStrategyTooltipLinkProps> = ({
                     maxHeight: 600,
                 }}
             >
-                <Typography component="span">
-                    {formatStrategyName(change.payload.name)}
```
Comment (thomasheartman) [inline frontend/src/component/changeRequest/ChangeRequest/SegmentTooltipLink.tsx:74]: Not sure why we were calling `formatStrategyName` here, but I suspect it was a copy-paste error that was left in.

File: frontend/src/component/changeRequest/changeRequest.types.ts
Comment (thomasheartman) [inline frontend/src/component/changeRequest/changeRequest.types.ts:1]: Update types to indicate that name is included.

**Review by kwasniew — APPROVED** [review]

**Pull request merged**